Lib/tempfile.py

```diff
@@ -100,6 +100,8 @@ def _infer_return_type(*args):
     for arg in args:
         if arg is None:
             continue
+        elif isinstance(arg, _os.PathLike):
```

Comment:
@lulouie No, path-like objects are only useful for things that represent paths. Both `prefix` and `suffix` are not paths themselves.